src/ops/delay.c

```diff
@@ -30,34 +30,37 @@ const tele_mod_t mod_DEL_R = MAKE_MOD(DEL.R, mod_DEL_R_func, 2);
 
 // common code to queue a delay shared between all delay ops
 // NOTE it is the responsibility of the callee to call tele_has_delays
-static void delay_common_add(scene_state_t *ss, exec_state_t *es, int16_t i,
+static bool delay_common_add(scene_state_t *ss, exec_state_t *es,
                              int16_t delay_time,
                              const tele_command_t *post_command) {
+    int16_t i = 0;
+
+    // 0 is the magic number for an empty slot.
+    // Be careful not to set delay.time[i] to 0 before calling this function.
+    while (ss->delay.time[i] != 0 && i != DELAY_SIZE) i++;
+
+    if (delay_time < 1) delay_time = 1;
```

Comment:
added a check for the delay time to the common code function. removed the check from the ```DEL``` op function, but it's still in the ```DEL.X``` and ```DEL.R``` because the wrap check would cause 32ms delays if the delay time param was negative.